wifiphisher/common/interfaces.py

```diff
@@ -623,13 +623,66 @@ def on_exit(self):
                 mac_address = adapter.original_mac_address
                 self.set_interface_mac(interface, mac_address)
 
+        if (hasattr(enable_network_manager, 'has_disable_nm') and
+                enable_network_manager.has_disable_nm):
+            # recover the network manager if we have disabled it past
+            enable_network_manager(True)
+
+
+def enable_network_manager(nm_enable):
+    """
+    Enable or Disable the NetworkManager
+
+    :param enable: Enable the network manager if True and disable the
+    network manager if False
+    :type enable: bool
+    :return None
+    :rtype: None
+    .. note: Disable the network manager when we don't want to access
+    internet via it. If there is no network manager dbus-service alive
+    accessing the dbus will cause exception. This exception can be safely
+    pass since when network manager is not running, we don't need to disable
+    and recover it back.
+    """
+
+    bus = dbus.SystemBus()
+    enable_network_manager.has_disable_nm = False
```

Comment:
Since I need to remember whether I have disabled the NetworkManager before. I only disable it when users are not giving the `-iI` option. When `network_manager.on_exit`, I can use this attribute to classify whether the NetworkManager has been disabled before.

if returning a value, this value must be passed to `network_manager.on_exit` and this change the signature of that function, I'm not so sure we want to do that. what do you think?